keps/498-GroupScheduling/README.md

```diff
@@ -0,0 +1,395 @@
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1: Group scheduling of all replicated jobs](#story-1-group-scheduling-of-all-replicated-jobs)
+  - [Example JobSet spec using group scheduling for all replicated jobs:](#example-jobset-spec-using-group-scheduling-for-all-replicated-jobs)
+    - [Story 2: Group scheduling of a specific replicated job](#story-2-group-scheduling-of-a-specific-replicated-job)
+  - [Example JobSet spec using group scheduling for all replicated jobs:](#example-jobset-spec-using-group-scheduling-for-all-replicated-jobs-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Proposed Group Scheduling API](#proposed-group-scheduling-api)
+  - [Constraints](#constraints)
+  - [Implmentation](#implmentation)
+    - [Example JobSet spec AFTER webhook injection:](#example-jobset-spec-after-webhook-injection)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+
+## Motivation
+
+Many users of managed K8s services on cloud providers make use of NAP (node auto provisioning) which creates node pools for pending/unschedulable pods, based on those pods requirements (i.e., CPU/memory requirements, GPU/TPU requirements, etc).
+
+Since node pool provisioning takes a variable amount of time, users are running into issues where the first slice finishes provisioning and pods land there and begin running, but eventually timeout before the other slices all finish provisioning and pods land there and become ready.
+
+For statically provisioned clusters, we have recommended users use Kueue to handle group scheduling of JobSets once sufficient resources are available.
+
+However, JobSets running independently (without Kueue) on dynmaically provisioned clusters, we need to support group
+scheduling semantics in order to avoid these timeout issues.
+
+### Goals
+
+- Enhance the JobSet API with options allowing the user to define a group scheduling policy.
+
+### Non-Goals
+
+- Upstream changes that would improve/simplify group scheduling support in JobSet.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1: Group scheduling of all replicated jobs
+
+As a user, in order to make efficient use of expensive accelerator (GPU/TPU) resources, I use node
+auto-provisioning to provision infrastructure on an as-needed basis when a pending workload requires
```

Comment:
the node auto-provisioner is a GKE specific. However, similar problem arise in any autoscaled environment.